x-pack/legacy/plugins/apm/public/components/app/Settings/AddSettings/AddSettingFlyout.tsx

```diff
@@ -88,6 +84,11 @@ export function AddSettingsFlyout({
   const hasCorrectDecimals = Number.isInteger(sampleRateFloat * 1000);
   const isSampleRateValid =
     sampleRateFloat >= 0 && sampleRateFloat <= 1 && hasCorrectDecimals;
+
+  if (!isOpen) {
+    return null;
+  }
```

Comment:
Good catch! Is it erroring out for these cases as well (where `useState` is called conditionally)?

Reply:
lint rule